DOCUMENTATION.md

```diff
@@ -0,0 +1,165 @@
+### prefer_match_file_name
+A `prefer_match_file_name` rule which warns about
+ mismatch between file name and declared element inside
+
+
+### prefer_conditional_expressions
```

Comment:
Can we add a top-level header and a Table Of Contents?

Reply:
oh, yeah, we discussed table of contents and i totally forgot to implement it. question of one commit!